File: model-client/src/main/java/org/modelix/model/operations/NoOp.java
package org.modelix.model.operations;

import org.modelix.model.api.IWriteTransaction;

public class NoOp extends AbstractOperation implements IAppliedOperation {


  @Override
  public IAppliedOperation apply(IWriteTransaction transaction) {
    return this;
  }

  @Override
  public IOperation transform(IOperation previous) {
    return null;
  }

  @Override
  public IOperation getOriginalOp() {
    return this;
  }

  @Override
  public IOperation invert() {
    return this;
  }

}

File: model-client/src/main/kotlin/org/modelix/model/operations/NoOp.kt
package org.modelix.model.operations

import org.modelix.model.api.IWriteTransaction

class NoOp : AbstractOperation(), IAppliedOperation {
    override fun apply(transaction: IWriteTransaction?): IAppliedOperation? {
        return this
    }

    override fun transform(previous: IOperation?): IOperation? {
        return null
    }

    override val originalOp: IOperation
        get() = this

    override fun invert(): IOperation? {
        return this
    }
}
